### PR TITLE
add SetParameter command

### PR DESCRIPTION
**Pull request by DeadByElpy**

### CODE REVIEW

File: src/constants/deviceParameters.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/**
+ * Device parameters. Used in SetParameter/GetParameter.
+ */
+
+export const INITIAL_DATA = 23;
+export const ABSOLUTE_DATA_STATUS = 24;
+export const INITIAL_DATA_MULTI_CHANNEL = 29;
```
[the remark this answers is not in the document]
Comment (DeadByElpy) [reply src/constants/deviceParameters.ts:7]: docs added

File: src/commands/downlink/SetParameter.ts [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    // returns full message - header with body
+    toBytes (): Uint8Array {
+        const {id, data} = this.parameters;
+        const buffer = new CommandBinaryBuffer(COMMAND_MAX_BODY_SIZE);
+
+        let parameterData;
+
+        buffer.setUint8(id);
+
+        if ( id === deviceParameters.INITIAL_DATA ) {
```
Comment (DarkPark) [inline src/commands/downlink/SetParameter.ts:186]: switch/case?

File: src/commands/downlink/SetParameter.ts [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    IParameterAbsoluteDataStatus;
+
+/**
+ * SetParameter command parameters
+ *
+ * @example
+ * import {constants} from 'jooby-codec';
+ * {id: constants.deviceParameters.INITIAL_DATA, data: {value: 2023, meterValue: 204, pulseCoefficient: 100}}
+ */
+interface IDownlinkSetParameterParameters {
```
Comment (DarkPark) [inline src/commands/downlink/SetParameter.ts:54]:
```suggestion
interface ISetParameterParameters {
```
?

File: tests/commands.test.ts [hunk 4]
```diff
@@ -27,56 +28,89 @@ const {uplink, downlink} = commands;
...
+    //         lrc: '41'
+    //     }
+    // },
     {
-        constructor: downlink.CorrectTime2000,
-        name: 'downlink command 0x0c:CORRECT_TIME_2000',
-        parameters: {sequenceNumber: 45, time: -120},
+        constructor: downlink.SetParameter,
+        name: 'downlink command 0x03:SET_PARAMETER',
+        parameters: {id: deviceParameters.ABSOLUTE_DATA_STATUS, data: {status: 1}},
```
Comment (DarkPark) [inline tests/commands.test.ts:64]: let's split to some lines

File: tests/commands.test.ts [hunk 5]
```diff
@@ -57,6 +58,42 @@ const downlinkCommands: TCommandList = [
...
+        },
+        hex: {
+            header: '03 0b',
+            body: '1d 01 00 00 01 92 80 00 00 07 f0',
+            lrc: 'a5'
+        }
+    },
     {
         constructor: downlink.SetTime2000,
         name: 'downlink command 0x02:SET_TIME_2000',
```
[the remark this answers is not in the document]
Comment (DeadByElpy) [reply tests/commands.test.ts:99]: try merge project with PR to got full context

File: src/commands/downlink/SetParameter.ts (hunk 2 above)
Comment (DeadByElpy) [reply src/commands/downlink/SetParameter.ts]: fixed

**Review by DarkPark — APPROVED** [review]

**Pull request merged**